### PR TITLE
Teleportation improvement

### LINKED ISSUE #301: Ability to limit which worlds crafts can teleport to
**Is your feature request related to a problem? Please describe.**
Currently, crafts can teleport between worlds, but there is currently no way to prevent crafts from teleporting to worlds that are otherwise off-limits to regular players

**Describe the solution you'd like**
Either in config.yml or craft-files, add a flag called `denyTeleportToWorlds` where crafts will be unable to teleport to worlds listed under that flag

### PR DESCRIPTION
**Pull request by eirikh1996**
### Describe in detail what your pull request acomplishes

In this pull request, I am adding the ability to disable teleport to certain worlds and adds teleportation cooldown

### Related issues
- Fixes #301 

### Checklist
- [x] Unit tests 
- [x] Proper internationalization
- [x] Compiled/tested

### CODE REVIEW

File: modules/api/src/main/java/net/countercraft/movecraft/craft/CraftType.java [hunk 1]
```diff
@@ -316,6 +317,8 @@ public CraftType(File f) {
         int dropdist = integerFromObject(data.getOrDefault("gravityDropDistance", -8));
         gravityDropDistance = dropdist > 0 ? -dropdist : dropdist;
         collisionSound = Sound.valueOf((String) data.getOrDefault("collisionSound", "BLOCK_ANVIL_LAND"));
+        disableTeleportToWorlds = stringListFromObject(data.getOrDefault("disableTeleportToWorlds", new ArrayList<>()));
```
Comment (TylerS1066) [inline modules/api/src/main/java/net/countercraft/movecraft/craft/CraftType.java:320]: I defer to CC's knowledge on the subject, but is an ArrayList the best data structure for this?

Comment (eirikh1996) [reply modules/api/src/main/java/net/countercraft/movecraft/craft/CraftType.java:320]: stringListFromObject converts a List to a string array

File: modules/api/src/main/java/net/countercraft/movecraft/craft/CraftType.java [hunk 2]
```diff
@@ -109,6 +108,8 @@
     @NotNull private final List<Material> harvesterBladeBlocks;
     @NotNull private final Set<Material> passthroughBlocks;
     @NotNull private final Set<Material> forbiddenHoverOverBlocks;
+    @NotNull private final String[] disableTeleportToWorlds;
```
Comment (cccm5) [inline modules/api/src/main/java/net/countercraft/movecraft/craft/CraftType.java:111]: Use a Set

File: modules/api/src/main/java/net/countercraft/movecraft/craft/CraftType.java (hunk 1 above)
Comment (cccm5) [reply modules/api/src/main/java/net/countercraft/movecraft/craft/CraftType.java:320]: I would store it in a set after, as the main use of `disableTeleportToWorlds` is checking if an element is inside it

**Comment by eirikh1996** [comment]
requested changes made @cccm5

**Pull request merged**